### ISSUE TITLE
Missing ability to refresh token in msal-node when acquired by acquireTokenByCode

### ISSUE THREAD

**Issue by naugtur** [AzureAD/microsoft-authentication-library-for-js#2155]
>> Please follow the issue template below. Failure to do so will result in a delay in answering your question.

## Library
- [ x ] `@azure/msal-node@1.0.0-alpha.4 `

## Description

There doesn't seem to be a way to refresh a token acquired like so:
```js
const cca = new msal.ConfidentialClientApplication(config);

(...)

const tokenRequest = {
    code: req.query.code,
    scopes,
    redirectUri,
  };

  cca
    .acquireTokenByCode(tokenRequest)
    .then(async (response) => {
```
Specific issues:  
1. response doesn't contain refresh token
2. there doesn't seem to be a way to get a token refresh from cache
3. when config contains a cachePlugin, it doesn't seem to be used.
4. `cca.getCacheManager` doesn't exist

I believe there is a gap in the logic here that would be solved by returning the refresh token so that it can be used in the `acquireTokenByRefreshToken` call later on.

I could gain access to a refresh token if I recreate a key and get it from `cca.getTokenCache()` but that doesn't seem like the intended way.

## MSAL Configuration

```js
const config = {
  auth: {
    clientId: "...",
    authority: "https://login.microsoftonline.com/common",
    clientSecret: "...",
  },
  cache: {
    cachePlugin
  },
  system: {
    loggerOptions: {
      loggerCallback(loglevel, message, containsPii) {
        console.log(message);
      },
      piiLoggingEnabled: true,
      logLevel: msal.LogLevel.Verbose,
    },
  },
};
const scopes = ["user.read", "files.readwrite.all", "sites.readwrite.all"];
const cca = new msal.ConfidentialClientApplication(config);
```

## Reproduction steps
use ConfidentialClientApplication in Node.js

## Expected behavior

There is a way to perform a refresh flow

## Browsers/Environment

Node.js

**Comment by technical-boy** [comment]
Hi @naugtur ! Could you please try using the Silent flow, which internally uses the refresh token to fetch new tokens as and when needed. The refresh token is not returned to the user by design, it is intended to be used only via cache during token expiry.

You can refer to our [Silent flow sample](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/msal-node-samples/silent-flow) for a practical example of how to use the `acquireTokenSilent` API. Specifically, look at [this file](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/92e7f3d7db0aa6ac81d3847b0b0b7a236bc90557/samples/msal-node-samples/silent-flow/index.js#L115) for the usage example.

**Comment by naugtur** [comment]
Thanks for the reference. I am aware of the silent flow and I decided I'd rather refresh explicitly. Here's why:

TL;DR: if there's more than one app instance, it's gonna be really hard to not fall into concurrency issues.

- msal is in charge of when to refresh the token
- msal does that based on state in memory and the cache I can persist
- from the outside, cache is a single blob I can load and save. I could share it between instances of my app, but as number of users increases, I'll keep waiting to fetch all auth cache from storage before each request (or at least before each refresh request) 
- there's no way to lock the account record while it's being refreshed, so if more than one instance attempts to refresh - one will refresh and the other will fail. I now need to resolve a conflict on the cache and get the one that failed to try again
- there's likely more situations where I'll need to resolve a conflict on the cache if there's multiple live apps using one cache

Overall the cache and its persistence is not designed to work for a deployment with multiple processes.  
I thought that's why the raw refresh function was exposed in the API.

I'm happy to discuss, provide more details, help design it a bit more or contribute some of it.
